iModelCore/ECDb/ECDb/ViewGenerator.cpp

```diff
@@ -871,6 +870,44 @@ BentleyStatus ViewGenerator::RenderRelationshipClassEndTableMap(NativeSqlBuilder
             sqlBuilder.Append(" AS INTEGER)");
         };
 
+    if (ClassViews::IsViewClass(*relationMap.GetRelationshipClass().GetTarget().GetAbstractConstraint()))

```

Comment:
@khanaffan this is another section to take a closer look. It makes it so the relationship classes in ECDbMeta do not need a view query. Instead this goes and generates a view query dynamically by searching for a navigation property. It seems to work fine and I like this mechanism a lot. But maybe you see some flaw with the approach.